### PR TITLE
Fix the static check miss in BN_hex2bn

### LINKED ISSUE #17298: the date in *bn will be updated in case of BN_FLG_STATIC_DATA flag  in  BN_hex2bn
1 BN_FLG_STATIC_DATA indicate the read-only data
void bn_set_static_words(BIGNUM *a, const BN_ULONG *words, int size)
{
    /*
     * |const| qualifier omission is compensated by BN_FLG_STATIC_DATA
     * flag, which effectively means "read-only data".
     */
    a->d = (BN_ULONG *)words;
    a->dmax = a->top = size;
    a->neg = 0;
    a->flags |= BN_FLG_STATIC_DATA;
    bn_correct_top(a);
}
2 the date in *bn will be updated in case of BN_FLG_STATIC_DATA flag  in  BN_hex2bn
int BN_hex2bn(BIGNUM **bn, const char *a)
{

    /* i is the number of hex digits */
    if (bn_expand(ret, i * 4) == NULL)  // in case of no expand
        goto err;

}

### PR DESCRIPTION
**Pull request by KanPlus**
Fix #17298

### CODE REVIEW

File: crypto/bn/bn_conv.c [hunk 1]
```diff
@@ -154,6 +154,10 @@ int BN_hex2bn(BIGNUM **bn, const char *a)
             return 0;
     } else {
         ret = *bn;
+        if (BN_get_flags(ret, BN_FLG_STATIC_DATA)) {
+            ERR_raise(ERR_LIB_BN, BN_R_EXPAND_ON_STATIC_BIGNUM_DATA);
```
Comment (t8m) [inline crypto/bn/bn_conv.c:158]: I'd use ERR_R_PASSED_INVALID_ARGUMENT instead.

Comment (KanPlus) [reply crypto/bn/bn_conv.c]: This error code will be better, Updated.

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
@KanPlus I'd suggest using `git commit --fixup` to commit the additional fixup commits. It helps when the PR is being merged.

**Comment by KanPlus** [comment]
> 



>

**Comment by KanPlus** [comment]
> @KanPlus I'd suggest using `git commit --fixup` to commit the additional fixup commits in future. It helps when the PR is being merged.

Thanks, I will

**Review by t8m — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
Merged, thanks for the fix.

**Pull request closed without merging**